Dockerfile

```diff
@@ -134,7 +134,7 @@ RUN \
 ADD config /etc/nginx/
 ADD *.sh /usr/bin/
 
-VOLUME /etc/nginx/conf.d /var/log/nginx
+VOLUME /etc/nginx/conf.d /etc/nginx/lua /var/log/nginx
```

Comment:
It's name as /etc/nginx/canary as well as this repo (nginx-canary) but it stores configuration for both canary and latest.
There are 4 lua scrips: router.lua, routerlib.lua, config.lua, and versions.lua. Both router.lua and routerlib.lua are static scripts with the logic to implement the canary-latest routing that import config.lua and versions.lua. The config.lua script is dependent on the project (it contains the domain for the cookies and other settings). The versions.lua script contains the version for each deployment group and it will be changed for every deployment.
I moved the versions.lua and config.lua to /etc/nginx/canary directory because it is possible to mount this volume with the content of these files, but still using the router scripts provided by the container. Due to lua import policy, I made symlinks in the /etc/nginx/lua directory to the versions and config scripts.